### PR TITLE
Fix false positive of in UnnecessaryParentheses

### PR DESCRIPTION
**Pull request by atulgpt**
Fixes https://github.com/detekt/detekt/issues/5631

### CODE REVIEW

**Review by BraisGabin — APPROVED** [review]

File: detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnnecessaryParenthesesSpec.kt [hunk 1]
```diff
@@ -322,6 +322,165 @@ class UnnecessaryParenthesesSpec {
...
+                val violation = ++(2.foo)
+                var a = 2.foo
+                val violation1 = (++a.value)
+                val violation2 = ((++a).value)
+                val violation3 = (++(a.value))
+                val violation4 = ++((a.value))
+            }
+        """.trimIndent()
+
+        assertThat(testCase.rule.lint(code)).hasSize(if (testCase.allowForUnclearPrecedence) 5 else 7)
```
Comment (cortinico) [inline detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnnecessaryParenthesesSpec.kt:434]: Can I ask you to split those tests in 2 and have two different test name? It makes for tests that are easier to understand and debug when they fail

Comment (atulgpt) [reply detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnnecessaryParenthesesSpec.kt]: Done

**Comment by atulgpt** [comment]
Hi @BraisGabin / @cortinico this PR is good for merge, right? Let me know if o/w

**Comment by atulgpt** [comment]
Hi @BraisGabin, here as well milestone is not set 🙂

**Comment by cortinico** [comment]
> Hi @BraisGabin, here as well milestone is not set 🙂

Added

**Pull request merged**